### DISCUSSION TITLE
Differences to siggo

### DISCUSSION THREAD

**Discussion by dataO1** [isamert/scli#157] [Q&A]
So i found this other signal cli [siggo](https://github.com/derricw/siggo) and i was wondering what are the differences. Both your implementation and siggo seem to use signal-cli.

**Comment by exquo** [comment]
One difference is: siggo sounds like a cooler name :)

Both are TUIs for Signal with a Vim-like UX. For the end-user the differences come down to the features, bugs and design choices of each. You can try using both and see which one has the features you need and the quirks you don't.

One of the reasons for this duplication of features and code is the developers' preferences. It's nice to be able to choose your tools, from the programming language to the UI framework, libraries, etc. It is also often easier to start a new codebase from scratch than to examine and modify an existing one. These are hobby projects for the people creating them, so they might not mind the effort. A developer's goal is not necessarily to have the most complete software of its kind, but to gain some experience and to play with the tools they want. 

There is another TUI client, [gurk](https://github.com/boxdot/gurk-rs), which uses a different signal library in Rust. And there are less direct ways to have a signal client in a terminal: use bridges to other protocols (e.g. matrix) or plugins to other programs (e.g. WeeChat). See [this list](https://github.com/exquo/signal-soft/wiki/Software-list) for the software that I'm aware of.

**Reply by dataO1** [reply]
Thanks for the info :)
